[17.0][MIG] fieldservice_agreement: Migration to 17.0

imlopes opened the pull request:
Depends on: 
agreement: 

- [x] https://github.com/OCA/agreement/pull/35
- [ ] https://github.com/OCA/agreement/pull/39
- [ ] https://github.com/OCA/agreement/pull/40


Adding new `fieldservice_agreement_serviceprofile` (did in a new commit❓ )

Removing dependency from `agreement_serviceprofile` on `fieldservice_agreement`  and adding a new glue module :
agreement_serviceprofile: https://github.com/OCA/agreement/pull/39

...

Commit:
[ADD] fieldservice_skill

[UPD] README.rst

Commit:
[11.0][ADD] fieldservice_agreement module

[UPD] README.rst

max3903 committed:
[IMP] Various improvements

Commit:
11.0 imp fieldservice_agreement (#76)

* [IMP] fieldservice_agreement

[UPD] README.rst

max3903 committed:
[IMP] Icon

max3903 committed:
[IMP] Show stat and improve filter

max3903 committed:
[IMP] Filtering

max3903 committed:
[IMP] fieldservice_agreement

max3903 committed:
[ADD] Merge fieldservice_equipment_agreement

osi-scampbell committed:
[IMP] Change Kanban Date

Commit:
[FIX]1. message_follower_ids needs to be inherited in object first 2. move views to agreement module 3. improve location heirarchy name construction

osi-scampbell committed:
[IMP] Fieldservice Account Bug Fix

murtuzasaleh committed:
[MIG][WIP][12.0] fieldservice_agreement

[UPD] README.rst

osi-scampbell committed:
[IMP] #118 Agreement Filter Location

osi-scampbell committed:
[IMP] #115 Worker Agreement Smart Button

[UPD] README.rst

max3903 committed:
[IMP] fieldservice_agreement

osi-scampbell committed:
[FIX] fieldservice_agreement (#232)

osi-scampbell committed:
[FIX] fieldservice_agreement: FSM Location Agreement Bug (#237)

[UPD] README.rst

kittiu committed:
[12.0][IMP] fieldservice_agreement, add unit test

osi-scampbell committed:
[IMP] Filter Out Templates

max3903 committed:
[FIX] fieldservice_agreement

max3903 committed:
[FIX] fieldservice_agreement

max3903 committed:
[FIX] fieldservice_agreement

[IMP] fieldservice_vehicle_stock

osi-scampbell committed:
[IMP] Improved Test Cases

[UPD] Update fieldservice_agreement.pot

Added translation using Weblate (Portuguese (Brazil))

Added translation using Weblate (Spanish (Chile))

Translated using Weblate (Spanish (Chile))

Currently translated at 100.0% (15 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/es_CL/

Added translation using Weblate (German)

Translated using Weblate (German)

Currently translated at 6.7% (1 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/de/

Translated using Weblate (German)

Currently translated at 26.7% (4 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/de/

Translated using Weblate (German)

Currently translated at 100.0% (15 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/de/

[ADD] Spanish (es) translation

Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (15 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/pt_BR/

Translated using Weblate (Spanish)

Currently translated at 100.0% (15 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/es/

Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (15 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/pt_BR/

Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (15 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/pt_BR/

Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (15 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/pt_BR/

Added translation using Weblate (Spanish (Argentina))

Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (15 of 15 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_agreement/es_AR/

patrickrwilson committed:
[14.0][MIG] fieldservice_agreement (Version 12.0 to 14.0)

[MIG] Black

[MIG] Added more coverage to tests

[MIG] Add contract to dependancy

Add maintainer

Update CONTRIBUTORS.rst

add eol

oca-travis committed:
[UPD] Update fieldservice_agreement.pot

OCA-git-bot committed:
[UPD] README.rst

hparfr committed:
Fix pre-commit

It will be better to properly fix by not using replace I guess

OCA-git-bot committed:
fieldservice_agreement 14.0.1.1.0

brian10048 committed:
[FIX] fieldservice_agreement: adapt access to ir.actions.* records

OCA-git-bot committed:
fieldservice_agreement 14.0.1.1.1

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (19 of 19 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice_agreement/it/

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-14.0/field-service-14.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice_agreement/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (19 of 19 strings)

Translation: field-service-14.0/field-service-14.0-fieldservice_agreement
Translate-URL: https://translation.odoo-community.org/projects/field-service-14-0/field-service-14-0-fieldservice_agreement/es_AR/

szekawong committed:
[MIG] fieldservice_agreement: Migration to 16.0

imlopes committed:
[IMP] fieldservice_agreement: pre-commit auto fixes

imlopes committed:
[17.0][MIG] fieldservice_agreement: Migration to 17.0

imlopes committed:
Add dependency file for tests purposes

ivantodorovich commented:
Can you take care of this :

> TODO: Add new fieldservice_agreement_serviceprofile ❓


Otherwise we're not supporting the same features that were before the migration.

imlopes committed:
[17.0[ADD] fieldservice_agreement_serviceprofile

As we're removing the dependency on `agreement_serviceprofile`, a new glue module is added to keep the same features than before.

imlopes commented:
> Can you take care of this :
> 
> > TODO: Add new fieldservice_agreement_serviceprofile ❓
> 
> Otherwise we're not supporting the same features that were before the migration.

I did on a new commit

max3903 commented:
@imlopes Can you split this pull request and have one for each module?

yankinmax commented on fieldservice_agreement/views/agreement_view.xml:
```diff
@@ -24,7 +24,7 @@
                     class="oe_stat_button"
                     icon="fa-laptop"
-                    context="{'default_agreement_id': active_id}"
+                    context="{'default_agreement_id': id}"
```
why not `active_id`?

imlopes replied:
I got from here : 

> If you are using active_id, active_ids or active_model in your views (for example, for a smart-button context), you should replace them by the corresponding good practices:
> 
>     id instead of active_id.
>     Hard-coded model name instead of active_model.
>     active_ids doesn't have direct replacement candidate. It will depend on what you want to achieve. It's a weird case though.

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0

yankinmax commented on fieldservice_agreement_serviceprofile/models/fsm_location.py:
```diff
@@ -0,0 +1,29 @@
+            agreements = self.env["agreement"].search(
+                [("fsm_location_id", "=", loc.id)]
+            )
+            ids = []
```
```
agreement_ids = agreements.ids
service_profiles = self.env['agreement.serviceprofile'].search(
    [('agreement_id', 'in', agreement_ids)]
)
ids = list(set(service_profiles.ids))
```

ivantodorovich commented:
Superseeded by:
- https://github.com/OCA/field-service/pull/1266
- https://github.com/OCA/field-service/pull/1267
